audformat/core/define.py

```diff
@@ -69,6 +69,7 @@ class TableStorageFormat(DefineBase):
     r"""Storage format of tables."""
     CSV = 'csv'
     PICKLE = 'pkl'
+    ALL = 'all'
```

Comment:
This is not very elegant.
Another solution would be to not add it here, but then we have to allow something like `typing.Union[str, TableStorageFormat]` as input for the `storage_format` argument.

Reply:
We could also support `storage_format=None` and in that case store in all formats?

Reply:
Yes, this owuld be another solution. The downside is that you would expect `storage_format=None` to be the default then, wouldn't you?

Reply:
Mhh, but it's also a case that will be only used in special occasion. Or you think this will be often used?

Reply:
Or maybe there is yet another solution: we could add a flag that - when turned on - checks if there is already a table in another format and in that case also stores the table in that format.

Reply:
Something like `update_other_formats=True`.

Reply:
I guess you are right. I changed it now to support `None` and removed `all`.

Reply:
What about the other solution? Maybe we should add it as well and set it to `True` by default. Then you will not end up with different versions of the same table even if you forget to set `storage_format=None`.

Reply:
+ it will only write out the other format if really needed

Reply:
Good point, so I reverted again.

We have now:
* 'all', stores to all available formats
* 'update' updates only existing database files. This means if a database contains only PKL files for some tables, also only those are updated. Or if you delete a CSV file manually it will not written again.

Reply:
Please squash when merging

Reply:
You implemented it in a slightly different way what I had in mind. 

As far as I see, with the current implementation `Table.save(path, 'csv')` will not update if a pickled version exists (and vice-versa). Which is exactly the case we'd like to avoid. On the other hand, `Table.save(path, 'update')` will not save the table if it does not exist, which might also be dangerous.

What I had in mind is adding `update` as an additional boolean flag. If it is set to `True` it will - in addition to the format that is selected - check if other formats of the table exist already save those as well. E.g.: `Table.save(path, 'csv', update=True)` will always save the table as CSV, but if there is already a pickled table, it will also save as PKL. I think we can then even remove `'all'` again, cause as long as the flag is turned on, we can never have the case that we have a CSV and PKL file that do not match.

Reply:
If I read something like `Table.save(path, 'csv', update=True)` I would assume it would look for a cSV file, and updates its content if needed. But not that it would touch files with a different format.

I think our current solution with several output format is still not optimal, but I also don't have a better idea at the moment.

Reply:
One solution would be that we always write all formats.

Reply:
Then let's call it `update_other_formats` as I proposed earlier. I really think that this ensures the behavior we want: save to a specific format, but ensure to update other formats if they already exists.

Reply:
I neither see the use-case for saving to all formats, nor to only update formats that exist (which in the worst case means not saving the table at all and the user is not aware of it).

Reply:
OK, `update_other_formats` is fine with me, I will update accordingly.

Reply:
I added `update_other_formats=True` and removed `'all'` and `'update'`.

I also added test that create PKL files that are different from CSV and updates them afterwards.
The test is still failing and I don't know why at the moment.

Reply:
I updated the description accordingly.